turbopack/crates/turbo-tasks-backend/src/backend/operation/connect_child.rs

```diff
@@ -52,9 +52,10 @@ impl ConnectChildOperation {
         };
 
         // Quick skip if the child was already connected before
-        if !new_children.insert(child_task_id) {
+        if new_children.insert(child_task_id, is_immutable).is_some() {
```

Comment:
I made it use only a static flag to know the perf difference of using a dynamic flag using codspeed. 

See https://github.com/vercel/next.js/pull/80506